v3/src/components/graph/components/graph.tsx

```diff
@@ -29,7 +29,7 @@ import {getDragAttributeId, IDropData} from "../../../hooks/use-drag-drop"
 import "./graph.scss"
 
 interface IProps {
-  graphController: GraphController
+  graphController: GraphController | undefined // TODO, better approach?
```

Comment:
@kswenson this. surprisingly, "works" (along with two `graphController?`s below)...but doesn't look quite right. I guess we need to work out how much of a `Graph` should or should not exist without a `GraphController`?

Reply:
See code review commit.